### ISSUE TITLE
Feature request: Option to disable automatic updates

### ISSUE THREAD

**Issue by nickpops98** [MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-#169]
Just a simple idea to allow the user to disable automatic updates. Some people may have slow internet speeds or a small amount of data and would not like to install the update. It's not a big deal clicking "Not Now" every time you start up the program but it would be nice to be able to disable it entirely.
Keep up the great work!

**Comment by MarshallOfSound** [comment]
Automatic updates are a huge part of this app, the ability to force users to keep up to date is the only thing that allows me to guarantee a good user experience.  Allowing users to turn this off when lots of them wouldn't understand the implication is to me a bad plan.

I have a local spike that detects network type and if you are connected to a mtered network it won't auto download.  But disabling them entirely is not something I would want to do.

Basically google changes their stuff so frequently that I have to be able to push updates out to people, or they're going to be complaining "XX YY ZZ" doens't work anymore.

If you can see a way around this while allowing auto updates to be turned off I am all ears :smile:

**Comment by FuzzAU** [comment]
I've found this too, I'm sometimes a bit lazy to update. Perhaps there can be a notification for higher priority updates that fix functionality broken due to a change on google music.

Otherwise it might be best to have a "don't remind me for X days" style option in the pop up for the updates.

**Comment by MarshallOfSound** [comment]
I'm closing this as `wontfix` because the problems you describe will be fixed by the moving to Squirrel Updater as described in #171 

Basically no more update notifications, Squirrel handles everything behind the scenes.  Doesn't even require you download the entire update, you can download "delta packages".

**Comment by eturk1** [comment]
just another perspective: every large trusted software company asks the user if they want to turn off automatic updates. we can even do this on Windows. truth is, updates can be unstable. we programmers are human and make mistakes. 

this was mainly a request of large corps. We you have 10,000 computers. One flawed auto update could mean a week of extra work load and angry people that can't get their job done. 

just a thought from a old guy that was programming in the 90's and doing massive regression testing before a small release. (eek!) LOL 

A harmony between being good hackers that shoot from the hip and humbly knowing we break stuff and make users waste hours on our error.

**Comment by MarshallOfSound** [comment]
@eturk1 Windows was a bad example, windows will now force updates on you.  Nothing you can do about it, for good reason.  Yes stuff gets broken, but if the only reason people have for not upgrading is "it might be broken" they really need to rethink how they use software.

Software always has the chance to be broken / have bugs but accepthing that and updating is the only way the devs shooting from the hip :+1: can find out about the problems.  If I wasn't very active on this repo I could see a valid concern that an auto update bug may render the player unusable for days if not weeks.  But I can successfully lay claim to fixing the one or two critical issues that have popped up in a matter of hours.

I hear the concern in regards to updates bringing in bugs, but I simply can't afford to leave users behind.  

Lets say hypothetically that GPM changes their ENTIRE structure.  Suddenly all my code starts failing, I need to be able to force that update through.  Because the UI might not even be able to launch.  This is the benefit of background updaters.  They don't require the main app to be working in order to update.  With GPM as variant as it is (changes occur at an insane rate) the auto updater just needs to be there.

It is worth pointing out though that `3.0.0` will have a squirrel enabled registry key.  So if you REALLLLLLLLLYYYYYYYYYYY want to turn off auto updates, you should be able to change the reg key to override it.  But I really don't see the need.

**Comment by eturk1** [comment]
great point on MS update option change with Win 10! :)

Win10 does have a new approach: One can defer what MS calls "upgrades" but not security patches. It's a finer grain take on the all or nothing that has been done by Adobe, etc for years.
<img width="685" alt="2016-02-02" src="https://cloud.githubusercontent.com/assets/525280/12766913/fd49c190-c9ba-11e5-95f0-355de12dc6ca.png">

Details of "defer" meaning here: 
http://windows.microsoft.com/en-us/windows-10/defer-upgrades-in-windows-10

**Comment by eturk1** [comment]
here's a thought on external component breaking the software, and thus users becoming upset at the app & devs, not knowing about the back end:

On start, check for updates, and put a short text for reason for update. If user continues without an update, it's their choice (or app shuts down, kill switch).

slick solution: update version # & short text on run:

scenario a) v2.1.2 Bug fixes...
 (user may or may not ask for update)
scenario b) v3.0 Update to support new Google Play Music ... (required)
 (user must update or app closes)

having beta ring, where users test all releases before release, solves much of the user challenge. alas, the joys of having a successful app with more users :)

**Comment by nickpops98** [comment]
@MarshallOfSound All good. Was just an idea but it makes perfect sense not to do it if it will cause problems down the road if users don't update. Would still be nice to maybe have a 'Disable Minor Updates' option but still force users to install major updates but I understand this could also cause problems. Regardless on what path you go down I love your work! Keep it up!

**Comment by Zipfer** [comment]
@nickpops98 
For me autoupdate was reason to stoup use this app because I spent my LTE traffic instead listening music.
